shell/common/platform_util_mac.mm

```diff
@@ -58,6 +58,28 @@
       stringByAppendingString:@".loginhelper"];
 }
 
+std::string OpenItemOnThread(const base::FilePath& full_path) {
+  DCHECK([NSThread isMainThread]);
+
+  NSString* path_string = base::SysUTF8ToNSString(full_path.value());
+  if (!path_string)
```

Comment:
We do not need to check result of `SysUTF8ToNSString`, if it returns nullptr then it is out of memory error, not because of invalid path passed.

The original code was quite ancient, and it did lots of unnecessary checks.